drivers/gpu/drm/bridge/cdns-mhdp.c

```diff
@@ -1791,14 +1921,33 @@ MODULE_FIRMWARE(FW_NAME);
 static int mhdp_remove(struct platform_device *pdev)
 {
 	struct cdns_mhdp_device *mhdp = dev_get_drvdata(&pdev->dev);
-	int ret;
+	unsigned int timeout = 10;
+	bool stop_fw = false;
+	int ret = 0;
 
 	drm_bridge_remove(&mhdp->bridge);
 
-	ret = cdns_mhdp_set_firmware_active(mhdp, false);
-	if (ret) {
-		dev_err(mhdp->dev, "Failed to de-activate DP\n");
-		return ret;
+wait_loading:
+	spin_lock(&mhdp->start_lock);
+	if (mhdp->hw_state == MHDP_HW_LOADING && timeout-- > 0) {
```

Comment:
if timeout is 1 when entering this if() block, you first check if it's > 0, which is true, then you subtract one from timeout, so it's now 0. Then you retry, and if the hw_state is now HW_READY, everything has gone fine. But as timeout is 0, you print timeout error below.